update_cpes.py

```diff
@@ -271,14 +329,20 @@ def update_cpes(xml_file, cpe_vp_map, r7_vp_map, deprecated_cves):
                     logging.error("Didn't find CPE type '%s' for '%s' '%s'", cpe_type, vendor, product)
                     continue
 
-                vendor = vendor.lower().replace(' ', '_').replace(',', '')
-                product = product.lower().replace(' ', '_').replace(',', '').replace('!', '%21')
                 if 'unknown' in [vendor, product]:
                     continue
 
                 if (vendor.startswith('{') and vendor.endswith('}')) or (product.startswith('{') and product.endswith('}')):
```

Comment:
A note: `*.product` can contain interpolations anywhere in the string. For example `{hw.product} Firmware` or `Thing  {hw.portCount}"`. I recommend adjusting the following as part of this PR as well:

```suggestion
               if (vendor.startswith('{') and vendor.endswith('}')) or '{' in product or '}' in product:
```